File: 2760-longest-even-odd-subarray-with-threshold/2760-longest-even-odd-subarray-with-threshold.py
class Solution:
    def longestAlternatingSubarray(self, nums: List[int], th: int) -> int:
        res =0
        cnt =0
        for i in range(len(nums)):
            if(nums[i]<=th):
                if(cnt >0 and nums[i]%2 != nums[i-1]%2):
                    cnt +=1
                elif (cnt >0 and nums[i]%2==nums[i-1]%2):
                    if(nums[i]%2 ==1):
                        cnt =0
                    else :
                        cnt =1
                elif (cnt ==0 and nums[i]%2==0):
                    cnt =1
            else :
                cnt =0
            res = max(res,cnt)
        return res

File: 2760-longest-even-odd-subarray-with-threshold/2760-longest-even-odd-subarray-with-threshold.cpp
class Solution {
public:
    int longestAlternatingSubarray(vector<int>& nums, int th) {
        int res =0,cnt =0;
        for(int i=0;i<nums.size();++i){
            if(nums[i]<=th){
                if(cnt and nums[i]%2 !=nums[i-1]%2)
                    ++cnt;
                else if(cnt and nums[i]%2 ==nums[i-1]%2){
                    if(nums[i]%2==0)cnt=1;
                    else 
                        cnt =0;
                }
                else if(nums[i]%2==0 and cnt ==0){
                    ++cnt;
                }
            }else
                cnt =0;
            res = max(res,cnt);
        }
        return res;
    }
};